app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -0,0 +1,152 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.GridCells
+import androidx.compose.foundation.lazy.LazyVerticalGrid
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.wallpapers.Wallpaper
+
+/**
+ * A grid of selectable wallpaper thumbnails.
+ *
+ * @param wallpapers Wallpapers to add to grid
+ * @param selectedWallpaper The currently selected wallpaper
+ * @param onWallpaperSelected Action to take when a new wallpaper is selected.
+ */
+@Composable
+fun WallpaperSettings(
+    wallpapers: List<Wallpaper>,
+    selectedWallpaper: Wallpaper,
+    onWallpaperSelected: (Wallpaper) -> Unit,
+) {
+    Surface(color = FirefoxTheme.colors.layer2) {
+        WallpaperThumbnails(
+            wallpapers = wallpapers,
+            selectedWallpaper = selectedWallpaper,
+            onWallpaperSelected = onWallpaperSelected
+        )
+    }
+}
+
+/**
+ * A grid of selectable wallpaper thumbnails.
+ *
+ * @param wallpapers Wallpapers to add to grid
+ * @param selectedWallpaper The currently selected wallpaper
+ * @param numColumns The number of columns that will occupy the grid.
+ * @param onWallpaperSelected Action to take when a new wallpaper is selected.
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun WallpaperThumbnails(
+    wallpapers: List<Wallpaper>,
+    selectedWallpaper: Wallpaper,
+    numColumns: Int = 3,
+    onWallpaperSelected: (Wallpaper) -> Unit,
+) {
+    LazyVerticalGrid(
+        cells = GridCells.Fixed(numColumns),
+        modifier = Modifier.padding(vertical = 30.dp, horizontal = 20.dp)
+    ) {
+        items(wallpapers) { wallpaper ->
+            WallpaperThumbnailItem(
+                wallpaper = wallpaper,
+                isSelected = selectedWallpaper == wallpaper,
+                onSelected = onWallpaperSelected
+            )
+        }
+    }
+}
+
+/**
+ * A single wallpaper thumbnail.
+ *
+ * @param wallpaper The wallpaper to display.
+ * @param isSelected Whether the wallpaper is currently selected.
+ * @param aspectRatio The ratio of height to width of the thumbnail
+ * @param onSelected Action to take when this wallpaper is selected.
+ */
+@Composable
+private fun WallpaperThumbnailItem(
+    wallpaper: Wallpaper,
+    isSelected: Boolean,
+    aspectRatio: Float = 1.1f,
```

Comment:
I have a question about how we come up with the default `1.1f`?

Reply:
I just did some math based on the designs from UX. They're pretty clearly not quite squares in the design, and using an aspect ratio allows us to dynamically resize based on screen size.

Reply:
I guess what is strange to me is that we just didn't specify the width/height (in dp) that is used in the figma. Since the dp should also just take care of of the different displays.